src/uu/expr/src/syntax_tree.rs

```diff
@@ -79,24 +80,18 @@ impl RelationOp {
             }
         };
         if b {
-            Ok("1".into())
+            Ok(NumOrStr::Num(BigInt::from(1)))
         } else {
-            Ok("0".into())
+            Ok(NumOrStr::Num(BigInt::from(0)))
         }
     }
 }
 
 impl NumericOp {
-    fn eval(&self, left: &AstNode, right: &AstNode) -> ExprResult<String> {
-        let a: BigInt = left
-            .eval()?
-            .parse()
-            .map_err(|_| ExprError::NonIntegerArgument)?;
-        let b: BigInt = right
-            .eval()?
-            .parse()
-            .map_err(|_| ExprError::NonIntegerArgument)?;
-        Ok(match self {
+    fn eval(&self, left: &AstNode, right: &AstNode) -> ExprResult<NumOrStr> {
+        let a: BigInt = left.eval()?.to_bigint()?;
+        let b: BigInt = right.eval()?.to_bigint()?;
```

Comment:
```suggestion
        let a = left.eval()?.to_bigint()?;
        let b = right.eval()?.to_bigint()?;
```
The annotations can be removed because they are no longer necessary for `parse`.